### PR TITLE
Adding cookie flush store api

### LINKED ISSUE #8416: Cookie is not saved on application close
* Electron version: 1.4.14
* Operating system: Windows

Hi!
I faced with issue described in this two comments: 
https://github.com/electron/electron/issues/6388#issuecomment-236498471
https://github.com/electron/electron/issues/6388#issuecomment-244483403
The main issue is fixed, but the session is lost if the app is closed immediately after login and setting the cookie.

### Expected behavior
Cookie should be saved.

### Actual behavior
When the app is closed after less then 10-15 seconds a cookie is set, it is not saved when the app is reopened.

### PR DESCRIPTION
**Pull request by juturu**
Fixes #8416

### CODE REVIEW

File: docs/api/cookies.md [hunk 1]
```diff
@@ -104,3 +104,7 @@ on complete.
 
 Removes the cookies matching `url` and `name`, `callback` will called with
 `callback()` on complete.
+
+#### `cookies.flushStore()`
```
Comment (MarshallOfSound) [inline docs/api/cookies.md]: It looks like this method takes a callback in the code, assuming that is called on completion.  This should probably be documented here as a parameter 👍

Comment (juturu) [inline docs/api/cookies.md]: Thanks @MarshallOfSound, updated

**Comment by deepak1556** [comment]
@juturu can you fix the cpplint issues with [clang-format](https://github.com/electron/electron/blob/master/docs/development/clang-format.md), CI builds are failing due to it.

**Comment by juturu** [comment]
@deepak1556 fixed the lint issue. Merge at your convenience.

File: atom/browser/api/atom_api_cookies.h [hunk 2]
```diff
@@ -53,6 +53,7 @@ class Cookies : public mate::TrackableObject<Cookies>,
   void Remove(const GURL& url, const std::string& name,
               const base::Closure& callback);
   void Set(const base::DictionaryValue& details, const SetCallback& callback);
+  void FlushStore(const base::Closure& callback);
```
Comment (kevinsawicki) [inline atom/browser/api/atom_api_cookies.h]: Could we add a test for this to `api-session-spec.js` that just verifies the callback is invoked, you don't need to assert more than that, but just to have basic coverage of this API.

**Comment by kevinsawicki** [comment]
Thanks for adding this @juturu 👍 🚀

**Pull request merged**